pyqtgraph/graphicsItems/ColorBarItem.py

```diff
@@ -35,33 +36,40 @@ class ColorBarItem(PlotItem):
     sigLevelsChanged = QtCore.Signal(object)
     sigLevelsChangeFinished = QtCore.Signal(object)
 
-    def __init__(self, values=(0,1), width=25, cmap=None, label=None,
+    def __init__(self, values=(0,1), width=25, colorMap=None, label=None,
                  interactive=True, limits=None, rounding=1,
-                 orientation='vertical', pen='w', hoverPen='r', hoverBrush='#FF000080' ):
+                 orientation='vertical', pen='w', hoverPen='r', hoverBrush='#FF000080', cmap=None ):
         """
-        Create a new ColorBarItem.
+        Creates a new ColorBarItem.
 
         ==============  ===========================================================================
         **Arguments:**
         values          The range of values as tuple (min, max)
         width           (default=25) The width of the displayed color bar
-        cmap            ColorMap object, look-up table is also applied to assigned ImageItem(s)
+        colorMap        ColorMap object, look-up table is also applied to assigned ImageItem(s)
         label           (default=None) Label applied to color bar axis
         interactive     (default=True) Handles are displayed to interactively adjust level range
         limits          Limits to adjustment range as (low, high) tuple, None disables limit
         rounding        (default=1) Adjusted range values are rounded to multiples of this values
-        orientation     (default='vertical') 'horizontal' gives a horizontal color bar
+        orientation     (default='vertical') 'horizontal' or 'h' gives a horizontal color bar.
         pen             color of adjustement handles in interactive mode
         hoverPen        color of adjustement handles when hovered over
         hoverBrush      color of movable center region when hovered over
         ==============  ===========================================================================
         """
         super().__init__()
+        if cmap is not None: 
+            warnings.warn(
+                "The parameter 'cmap' has been renamed to 'colorMap' for clarity. "
+                "The old name will no longer be available in any version of PyQtGraph released after July 2022.",
+                DeprecationWarning, stacklevel=2
+            )
+            colorMap = cmap
         self.img_list  = [] # list of controlled ImageItems
         self.values    = values
-        self.cmap      = cmap
+        self.cmap      = colorMap
```

Comment:
Though to be in proper Qt style, `self.colorMap` should be a _method_ that returns the color map. So we actually want to store the value in `self._colorMap`, have the deprecated property for `cmap`, and an accessor method `self.colorMap()`.

```python
def colorMap(self):
    return self._colorMap

@property
def cmap(self):
    warnings.warn('deprecated!', DeprecationWarning, stacklevel=2)
    return self._colorMap
```

Reply:
Dear @outofculture , I didn't even consider that somebody might want to get the color map *back* from the ColorBarItem, but you are right, we should have that. While retrieving this as `some_color_map_object.cmap` was never documented, this is indeed a good way to remove that with a proper deprecation period.

Added, and I am going to look this up here the next time I need it 👍